src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs

```diff
@@ -0,0 +1,319 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.Backup.Model;
+using Microsoft.Azure.Commands.Sql.Database.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Sql.Database_Backup.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Get, "AzureRmSqlDatabaseLongTermRetentionBackup", DefaultParameterSetName = LocationNameSet, SupportsShouldProcess = true)]
+    public class GetAzureRmSqlDatabaseLongTermRetentionBackup : AzureSqlDatabaseLongTermRetentionBackupCmdletBase
+    {
+        /// <summary>
+        /// Parameter set name for backup name.
+        /// </summary>
+        private const string BackupNameSet = "BackupName";
```

Comment:
In your cmdlet review you only had 6 parameter sets but here you added two more - why do you need these as separate parameter sets?  It seems that the first four could really just be two.

Reply:
There may be a way to do this I couldn't find but it's tricky because if I combined all but the first out of the first four, there is a dependency that needs to happen (ie - if you provide server name, you need to provide location name). If there is a way to do that and it's better, I can make the change.